migrate/live-migration/live-migration-from-timescaledb.md

```diff
@@ -0,0 +1,153 @@
+transactions from your source database to Timescale.
+
+You should provision a dedicated instance to run the migration steps from.
+Ideally an AWS EC2 instance that's in the same region as the Timescale target service.
+For an ingestion load of 10,000 transactions/s, and assuming that the historical
+data copy takes 2 days, we recommend 4 CPUs with 4 to 8 GiB of RAM and 1.2 TiB of storage.
+
+<SourceTargetNote />
+
+In detail, the migration process consists of the following steps:
+1. Set up a target database instance in Timescale.
+1. Prepare the source database for live migration.
+1. Run the live migration image.
+1. Validate the data in target database and promote as new primary.
+
+<GettingHelp />
+
+<StepOne />
+
+## 2. Prepare the source database for live migration
+
+<LiveMigrationStep2 />
+
+<Highlight type="important">
+TimescaleDB 2.12 and above allows setting `REPLICA IDENTITY` on hypertables. However,
+if you're using a version below 2.12, it’s important that your hypertables have primary keys.
+If they don't, you will need to upgrade your TimescaleDB instance. This is because, without a
+primary key or `REPLICA IDENTITY`, operations such as `DELETE` and `UPDATE` will not be replicated,
+which will impact the consistency of your data post migration.
+</Highlight>
+
+## 3. Run live migration image
+
+First, set the database URIs as environment variables:
+
+```sh
+export SOURCE=""
+export TARGET=""
+```
+
+Next, download and run the live migration image:
+
+```sh
+docker run --rm -it \
+  -e PGCOPYDB_SOURCE_PGURI=$SOURCE \
+  -e PGCOPYDB_TARGET_PGURI=$TARGET \
+  timescale/live-migration:0.0.1
+```
+
+This command may take a while to complete. It’s recommended to run it in the background
+or use terminal multiplexers like `screen` or `tmux`.
+
+The command will take a snapshot of your source database, migrate existing data to the
+target, and stream live transactions from the source to the target. During this process,
+it will display the lag between the source and target databases in terms of WAL offset size.
+
+```sh
+[WATCH] Source DB - Target DB => 126MB
+```
+
+When the lag between the source and target database is less than 30 megabytes, it will
+start `ANALYZE` on the target database. This updates statistics in the target database,
+which is necessary for optimal querying performance in the target database. Wait for
+`ANALYZE` to complete.
+
+<Highlight type="note">
+Application downtime begins here.
+</Highlight>
+
+When you're _ready to take downtime_ (i.e., stop normal operations) on your production
+applications that are dependent on your source database and the _lag between the databases is below 30 megabytes_,
+you should stop your applications that perform DML queries on your source database.
+This is the downtime phase and will last until you have completed the validation step (4).
+Be sure to go through the validation step (4) and prepare the validation queries before
```

Comment:
What are the `validation queries`? If we are going to mention something like this, we should explain it further, maybe some examples.

Reply:
Validation queries is a generic term that has been described in step 4. The point here is to basically give a hint that go through the step 4 before continuing here just so that the user is aware. We do mention about some examples in step 4, like `You could compare the number of rows or an aggregate of columns to validate that the target database matches with the source.`